src/PostgresOutbox/Serialization/ITypeResolver.cs

```diff
@@ -0,0 +1,32 @@
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
+
+namespace PostgresOutbox.Serialization;
+
+public interface ITypeResolver
+{
+    Type Resolve(string value);
+    (string, JsonTypeInfo) Resolve(Type type);
+    JsonSerializerContext SerializationContext { get; }
+}
+
+public class TypeResolver(JsonSerializerContext serializationContext, INamingPolicy? namingPolicy=default): ITypeResolver
+{
+    public JsonSerializerContext SerializationContext { get; } = serializationContext;
+    private static readonly Dictionary<string, Type> TypeDictionary = [];
+    private static readonly Dictionary<Type, JsonTypeInfo> TypeInfoDictionary = [];
+
+    public  TypeResolver WhiteList<T>() where T:class
```

Comment:
**QUESTION:** Does that mean that each message type should be registered explicitly? I understand the security concern, but I'm not sure if it wouldn't be good to add an option to register the default resolution strategy for unresolved types. Then user could add it

Reply:
We could provide different strategies here. 
An even better approach(I'm adopting in a production system) is to leverage on [publication filter](https://www.postgresql.org/docs/current/logical-replication-row-filter.html#LOGICAL-REPLICATION-ROW-FILTER-RESTRICTIONS) capabilities.

Reply:
Added:
- https://github.com/oskardudycz/PostgresOutboxPatternWithCDC.NET/issues/12
- https://github.com/oskardudycz/PostgresOutboxPatternWithCDC.NET/issues/13